plugins/SLURM-Integration/ufm_slurm_base.py

```diff
@@ -107,12 +109,35 @@ def create_sharp_allocation(self, job_id, job_nodes):
             logging.error("Failed to allocate job's node %s to pkey::: Error==> %s" % (job_nodes, exc))
 
     def delete_sharp_allocation(self, job_id):
-        try:
-            logging.info("Delete SHArP allocation with app_id: %s" % job_id)
-            response = self.ufm._delete_sharp_allocation(self.server, self.session, self.auth_type, job_id)
-            logging.info("Request Response: %s" % str(response))
-        except Exception as exc:
-            logging.error("Failed to Delete SHArP allocation with app_id %s ::: Error==> %s" % (job_id, exc))
+        left_retries = self.num_of_retries
+        while True:
+            try:
+                logging.info(f"Attempting to delete sharp reservation with app_id: {job_id}")
+                response = self.ufm._delete_sharp_allocation(self.server, self.session, self.auth_type, job_id)
+                if response.status_code == http.client.NOT_FOUND:
+                    logging.warning(f"Deleting sharp reservation failed, sharp reservation with app_id: "
+                                    f"{job_id} is not found!")
+                    break
+                if response.status_code == http.client.NO_CONTENT:
+                    logging.info(f"Deleting sharp reservation with app_id: {job_id} completed successfully.")
+                    break
+                else:
+                    logging.error(f"Deleting sharp reservation with app_id: {job_id} failed! "
+                                  f"status_code: {response.status_code}, response: {response.text}")
+                    if self.num_of_retries == 0:
```

Comment:
looks like bug, why checking self.num_of_retries but decrementing left_retries?
left_retries should be used all over the func

Reply:
plz add test for func

Reply:
No, it's not a bug. During the initialization, I set left_retries to self.num_of_retries because I want to maintain the original value of self.num_of_retries without decrementing it. This is necessary to handle the scenario where self.num_of_retries is set to 0, indicating that I need to retry the deletion indefinitely.

Reply:
so plz put comment describing that check if infinite or compare to constant == INFINITE
also, too many nested ```ifs```, plz simplify
why need ``else``` after ```break```?

Reply:
I will add comment as you requested and try to simplify the ifs as much as I can